lxd/task/group.go

```diff
@@ -60,16 +58,17 @@ func (g *Group) Start(ctx context.Context) {
 		go func(i int) {
 			task.loop(ctx)
 
-			// Ensure running map is updated before wait group Done() is called.
-			g.mu.Lock()
-			defer g.mu.Unlock()
-
 			if g.running != nil {
```

Comment:
AFAIK checking if map is not nil should be concurrent-safe, because this is essentially just checking whether the pointer to a map is not nil. 

However, I just wanted to point that out in case I have missed some edge case that may occur (maybe initializing a map while some routine is reading from the map)?

Otherwise, LGTM.

Reply:
To be safe I've refactored this to add a NewGroup function that always initialises the running map so we don't have to worry about concurrently checking if its initialised or not.